File: 02_ProgrammingFundamentals/_03_Arrays/_02_Exercise/_04_ArrayRotation.java
package _02_ProgrammingFundamentals._03_Arrays._02_Exercise;

import java.util.Arrays;
import java.util.Scanner;

public class _04_ArrayRotation {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input values
        int[] numbers = Arrays.stream(scan.nextLine().split("\\s+")).mapToInt(Integer::parseInt).toArray();
        int n = Integer.parseInt(scan.nextLine());

        for (int i = 0; i < n; i++) {
            int numberInTheFront = numbers[0];
            int[] changingArray = new int[numbers.length];

            for (int j = 0; j < numbers.length - 1; j++) {
                changingArray[j] = numbers[j + 1];
            }

            changingArray[changingArray.length - 1] = numberInTheFront;
            numbers = changingArray;
        }

        // Print result
        Arrays.stream(numbers).forEach(e -> System.out.print(e + " "));

        scan.close();
    }
}

File: 24_C#Fundamentals/_03_Arrays/_02_Exercise/_04_ArrayRotation/_04_ArrayRotation.cs
using System;

public class _04_ArrayRotation
{
    public static void Main(string[] args)
    {
        // Read input values
        int[] numbers = Console.ReadLine().Split().Select(int.Parse).ToArray();
        int n = int.Parse(Console.ReadLine());

        for (int i = 0; i < n; i++)
        {
            int numberInTheFront = numbers[0];
            int[] changingArray = new int[numbers.Length];

            for (int j = 0; j < numbers.Length - 1; j++)
            {
                changingArray[j] = numbers[j + 1];
            }

            changingArray[changingArray.Length - 1] = numberInTheFront;
            numbers = changingArray;
        }

        // Print result
        Console.WriteLine(string.Join(" ", numbers));
    }
}
